### PR TITLE
[#IOPID-1763]  Add `session-manager` app

### PR DESCRIPTION
**Pull request by gquadrati**
## Description ##

This PR introduced the new session-manager app and the CI pipeline

### CODE REVIEW

File: apps/session-manager/.node-version [hunk 1]
```diff
@@ -0,0 +1 @@
+20.10.0
```
Comment (arcogabbo) [inline apps/session-manager/.node-version:1]: the monorepo has `20.12.2` as node version. could we also put it here?

File: apps/session-manager/package.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{
+    "name": "@pagopa/io-session-manager",
+    "version": "0.0.1",
+    "description": "API for IO session",
+    "main": "./dist/main.js",
```
Comment (arcogabbo) [inline apps/session-manager/package.json:5]: is this needed? what is the difference with `dist/server.js`

File: apps/session-manager/package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        "lint": "yarn lint:check --fix",
+        "lint:check": "eslint . -c .eslintrc.js --ext .ts,.tsx --cache",
+        "test": "vitest run",
+        "test:watch": "vitest"
+    },
+    "devDependencies": {
+        "@pagopa/typescript-config-node": "*",
+        "@types/express": "^4.17.21",
+        "@types/node": "~20.10.0",
+        "dependency-check": "^4.1.0",
```
Comment (arcogabbo) [inline apps/session-manager/package.json:24]: at the moment this is not used. should we add a `postbuild` script as follows? `"postbuild": "dependency-check package.json --no-dev --missing ./dist/**/*.js",`

File: apps/session-manager/package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    "dependencies": {
+        "@azure/functions": "^3.2.0",
+        "@pagopa/io-functions-commons": "^28.21.0",
+        "@pagopa/io-spid-commons": "^13.4.0",
+        "@pagopa/openapi-codegen-ts": "^13.1.0",
+        "@pagopa/ts-commons": "^12.6.0",
+        "express": "^4.19.2",
+        "fp-ts": "^2.16.5",
+        "io-ts": "^2.2.21",
+        "vitest": "^1.5.0"
```
Comment (arcogabbo) [inline apps/session-manager/package.json:42]: should be a `devDependency`

File: .github/workflows/code_review.yaml
Comment (arcogabbo) [inline .github/workflows/code_review.yaml:1]: could we also add the PR checker github action or is it out of scope? like [this](https://github.com/pagopa/io-app-email-templates/blob/main/.github/workflows/pr-title-linter-and-linker.yaml) one for example

File: apps/session-manager/.node-version (hunk 1 above)
Comment (gquadrati) [reply apps/session-manager/.node-version]: Done in [[#IOPID-1763] add eslint + fix based on code review](https://github.com/pagopa/io-auth-n-identity-domain/pull/1/commits/95787ff45a3ddbfeccbb0f616623810c957e883f)

File: apps/session-manager/package.json (hunk 2 above)
Comment (gquadrati) [reply apps/session-manager/package.json]: Removed in [[#IOPID-1763] add eslint + fix based on code review](https://github.com/pagopa/io-auth-n-identity-domain/pull/1/commits/95787ff45a3ddbfeccbb0f616623810c957e883f)

File: apps/session-manager/package.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
+{
+    "name": "@pagopa/io-session-manager",
+    "version": "0.0.1",
+    "description": "API for IO session",
+    "main": "./dist/main.js",
+    "license": "MIT",
```
[the remark this answers is not in the document]
Comment (gquadrati) [reply apps/session-manager/package.json]: Removed in [[#IOPID-1763] add eslint + fix based on code review](https://github.com/pagopa/io-auth-n-identity-domain/pull/1/commits/95787ff45a3ddbfeccbb0f616623810c957e883f)

File: apps/session-manager/package.json (hunk 3 above)
Comment (gquadrati) [reply apps/session-manager/package.json]: Removed for now in [[#IOPID-1763] add eslint + fix based on code review](https://github.com/pagopa/io-auth-n-identity-domain/pull/1/commits/95787ff45a3ddbfeccbb0f616623810c957e883f)

File: apps/session-manager/package.json (hunk 4 above)
Comment (gquadrati) [reply apps/session-manager/package.json]: Right, thanks! Moved in devDependency in [[#IOPID-1763] add eslint + fix based on code review](https://github.com/pagopa/io-auth-n-identity-domain/pull/1/commits/95787ff45a3ddbfeccbb0f616623810c957e883f)

File: .github/workflows/code_review.yaml
Comment (gquadrati) [reply .github/workflows/code_review.yaml:1]: Added in [[#IOPID-1763] add eslint + fix based on code review](https://github.com/pagopa/io-auth-n-identity-domain/pull/1/commits/95787ff45a3ddbfeccbb0f616623810c957e883f)

**Review by arcogabbo — APPROVED** [review]

**Review by BurnedMarshal — APPROVED** [review]
LGTM

**Pull request merged**